src/ripple/app/misc/NetworkOPs.cpp

```diff
@@ -474,16 +471,18 @@ class NetworkOPsImp final
     void onStop () override
     {
         mAcquiringLedger.reset();
-        m_heartbeatTimer.cancel();
-        m_clusterTimer.cancel();
+
+        if (heartbeatTimer_)
+            heartbeatTimer_->cancel();
+        if (clusterTimer_)
+            clusterTimer_->cancel();
 
         stopped ();
```

Comment:
I believe there's a race condition here. Consider if the completion handler is already running, and we call `stopped` just before the completion handler calls `addJob`. Now we have incorrectly signaled that we have stopped, but then we continue to use resources.